### PR TITLE
List Cilium's extended supported features 

### PR DESCRIPTION
**Pull request by xtineskim**
**What type of PR is this?**
/kind cleanup 

**What this PR does / why we need it**:

Related to https://github.com/kubernetes-sigs/gateway-api/issues/2550 (and somewhat: https://github.com/kubernetes-sigs/gateway-api/issues/2740. Guidelines on what fields would be required in the conformance report will help with parsing displayed results!)

**Which issue(s) this PR fixes**:
Other implementations provided listing out extended supported features, adding in Cilium's so that it can help with how conformance reports are displayed. 

Fixes # n/a

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: conformance/reports/v1.0.0/cilium.yaml [hunk 1]
```diff
@@ -24,6 +24,19 @@ profiles:
         Passed: 12
         Skipped: 0
       summary: ""
+      supportedFeatures:
```
Comment (sayboras) [inline conformance/reports/v1.0.0/cilium.yaml:27]: Thanks for adding this, I just update gateway-api version in cilium/cilium to pick up the new format.

Here is the extra block in new report generated by CI https://github.com/cilium/cilium/actions/runs/7718271763

```
    supportedFeatures:
    - HTTPRouteBackendTimeout
    - HTTPRouteHostRewrite
    - HTTPRouteMethodMatching
    - HTTPRouteParentRefPort
    - HTTPRoutePathRedirect
    - HTTPRoutePathRewrite
    - HTTPRoutePortRedirect
    - HTTPRouteQueryParamMatching
    - HTTPRouteRequestMirror
    - HTTPRouteRequestMultipleMirrors
    - HTTPRouteRequestTimeout
    - HTTPRouteResponseHeaderModification
    - HTTPRouteSchemeRedirect
```

Comment (sayboras) [reply conformance/reports/v1.0.0/cilium.yaml:27]: Apart from sorting order, we can add HTTPRouteParentRefPort into this list.

Comment (xtineskim) [reply conformance/reports/v1.0.0/cilium.yaml:27]: gotcha, updated it now. Thank you for the update!

**Review by sayboras — APPROVED** [review]
Thanks and LGTM.

Just a note for other reviewers, the report is basically generated from cilium CI. Here is the recent run.

https://github.com/cilium/cilium/actions/runs/7724870579/artifacts/1208443603

**Comment by robscott** [comment]
Looks like just a timeout on the failing test.

/retest

**Comment by robscott** [comment]
@xtineskim it looks like that latest failure was actually not a flake:

```
Successfully installed pathspec-0.12.1 pyyaml-6.0.1 yamllint-1.29.0
./conformance/reports/v1.0.0/cilium.yaml
  28:7      error    wrong indentation: expected 8 but found 6  (indentation)
```

Hopefully once that's updated this should be good to go.

**Comment by robscott** [comment]
Thanks @xtineskim!

/lgtm

**Pull request merged**